Simplified/AxisService/NYPLAxisServiceAdapter.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  NYPLAxisServiceAdapter.swift
+//  Open eBooks
+//
+//  Created by Raman Singh on 2021-06-23.
+//  Copyright © 2021 NYPL Labs. All rights reserved.
+//
+
+import Foundation
+import NYPLAxis
+
+@objc protocol NYPLBookDownloadBroadcasting {
+  func downloadProgressDidUpdate(to progress: Double, forBook book: NYPLBook)
+  func failDownloadWithAlert(forBook book: NYPLBook)
+  func replaceBook(_ book: NYPLBook,
+                   withFileAtURL sourceLocation: URL,
+                   forDownloadTask downloadtask: URLSessionDownloadTask) -> Bool
+}
+
+@objc
+class NYPLAxisBookDownloadAdapter: NSObject {
+  
+  private let axisService: NYPLAxisService
+  private let mediator: NYPLAxisBookDownloadMediator
+  
+  @objc
+  init?(downloadTask: URLSessionDownloadTask,
+       book: NYPLBook,
+       downloadBroadcaster: NYPLBookDownloadBroadcasting,
+       fileURL: URL) {
+    
+    let mediator = NYPLAxisBookDownloadMediator(
+      with: downloadBroadcaster, book: book, downloadTask: downloadTask)
+    let errorLogsAdapter = NYPLAxisErrorLogsAdapter()
+    
+    guard let service = NYPLAxisService(
+            delegate: mediator, fileURL: fileURL,
+            forBookWithIdentifier: book.identifier,
+            downloader: NYPLAxisContentDownloader(),
+            sha: book.identifier.sha256(), errorLogger: errorLogsAdapter,
+            axisXMLCreator: NYPLAxisXMLCreator())
+    else {
+      return nil
+    }
+    
+    self.mediator = mediator
+    self.axisService = service
+  }
+
+  @objc
+  func downloadBook() {
+    axisService.fulfillAxisLicense()
```

Comment:
fulfilling the license will basically trigger all the other parts of the "book download" process? such as metadata, container, assets etc

Reply:
yup

Reply:
do you mind adding a comment about that please?